### DISCUSSION TITLE
Failing to Host TCP/IP Game, Change Difficulty

### DISCUSSION THREAD

**Discussion by jh86** [isid0re/SoloLeveling#382] [Q&A]
So I made 8 Characters yesterday that I wanted to run together eventually. Everything started out fine, I selected the Host TCP/IP Mode and entered the pertinent information for the Characters to be created and have a build. The Sorceress and Assassin were the first to clear Normal but when they went back to create a Nightmare game they were unable to select the difficulty at the entry screen, similar issue to when they are joining a TCP/IP game and they freeze. One by one, the profiles began to have the error D2BS is not responding... starting profile.

I fiddled around with a few things and then eventually reinstalled. I remade the profiles and added the randomly generated names to the profile appropriately. They were identical in all but the build section and name however only 5 or the 8 worked! The Sorceress, Assassin and 1 Paladin who was not finished normal yet refused to open and repeated the loop of D2BS is not responding. Yet at the very same time 5 were levelling and working hard.

I woke up this morning and they had all progressed to this error again. I am wondering if anyone has dealt with this and if it is indeed possible to run Sololeveling in TCP/IP mode properly. It gets to the home screen and won't function. If I switch the starting script to mule for example and manually log in, it crashes anyways, as if the program is stuck in a set loop of inaction regardless of if it's paused or trying to run. I just hit Home button and it says cannot find x.dbj, cannot find starter.dbj and closes.

**Comment by isid0re** [comment]
Check the TCP ip settings on the tcp portion of the entry scripts. There might be a conflict issue with the ip portion preventing it from game entry. As there is a tcp host and tcp client entry onfor 

 It might be worth testing running them on tcp via ip labeling as:
127.0.0.1
Localhost
192.168.x
Or which ever internal ip network you are using 

I'll do the same and let's post findings on this thread so we can debug the situation more.

**Reply by jh86** [reply]
Just got home from work, will be tinkering tonight and I will post anything helpful, thank you for the reply!

**Reply by jh86** [reply]
I completely reinstalled everything again and all profiles were working. After a bit of wifi drop-out the profiles started to fail. I had a look through to find the area you suggested but I was having difficulties. I did not see any spot to enter an IP besides this one section in d2botsololevelingentry.dbj

I am not familiar with how to add the IP, I tested it out adding in brackets, quotations and quotations in brackets directly after case41: (located at line 1097)

break;
	case 41: // TCP/IP Game - Enter Host IP
		try {
			login(me.profile);
		} catch (e) {
		//
		}

		break;

Am I barking up the wrong tree? Can you advise where I can edit the IP address properly?

**Reply by jh86** [reply]
![troubleshooting](https://user-images.githubusercontent.com/51974054/152658131-786d963e-cfbe-4a1c-8cac-8f45380b3e8f.png)

**Reply by jh86** [reply]
This is what's displayed when I hit the home button before the generated game closes after approximately 10 seconds. It can't find the profile anymore and it also cannot find starter.dbj. If I swap this to any other entry script it has the exact same problem. In regular kolbot there's an editable profile that appears but that isn't so with sololeveling. It's very strange that after working perfectly it gets into a loop where it can't find the profile nor the starter. Is it specifically looking for a file called starter.dbj? This file does not exist! After reinstalling it was completely fine in selecting nightmare for those profiles that met the requirements so we can wipe that issue off the table.

**Reply by isid0re** [reply]
Ouch that's a corrupt ini file. But here is the easy fix so you dont have to reinstall.

If you installed kolbot via svn. Then go to the kolbot/d2bs directory and look for the d2bs.ini and right click and revert. 

If you zip installed then just download and replace the d2bs.ini file.

**Reply by jh86** [reply]
So that completely fixed the probelm, replacing the corrupt ini file - however the question is how have I corrupted the ini file 3 times now? I will keep an eye out and see how long before I start crashing again.

**Reply by isid0re** [reply]
Double check your dependencies. For kolbot make sure you are using the x86 distributable and not an a64 distributable. That may play a part. Kolbot is an older software and is build off x86 architecture. The other causes I'm aware of are windows updates and improper shutdowns while kolbot is running can corrupt the ini.

**Comment by isid0re** [comment]
I've started tinkering and I agree the tcp section doesn't clearly make things easy to edit. I originally incorporated portions of the blizzhackers repo tcp host entry script to enable compatibility with tcp play for solo. I made a modified entry to test and see if changing the host ip makes a difference on the game. I'll post findings soon

**Reply by jh86** [reply]
So I think I've found the smoking pistol. Had an error tonight and it prompted me to check the exceptions log. Here's an excerpt (ini became corrupted after this)

2022-02-05 8:50 PM
System.Net.HttpListenerException (0x80004005): Failed to listen on prefix 'http://127.0.0.1:8080/' because it conflicts with an existing registration on the machine.
   at System.Net.HttpListener.AddAllPrefixes()
   at System.Net.HttpListener.Start()
   at D2Bot.WebServer.Listen()

**Reply by jh86** [reply]
Well, I started having this problem again. I checked the ini file referred to earlier in this thread and it surely did not seem corrupted. This got me digging a bit further. It appears that there's trouble with port 8080, maybe because of limedrop. I went ahead and did a "netstat -a" in my command prompt and decided to switch to port 6364. I opened up kolbot > libs > modules > to http. In there I removed "80" from line 17 and changed it to 6364. I rebooted the d2bot.exe and everything loaded up fine! Will update if it stops working again.

**Reply by isid0re** [reply]
That's an interesting find. I never even considered it might have been a port issue
